packages/server/src/handlers/PostHandler.ts

```diff
@@ -178,13 +197,18 @@ export class PostHandler extends BaseHandler {
       if (created.offset !== Number.parseInt(upload_length as string, 10)) {
         const creation = new Date(upload.creation_date)
         // Value MUST be in RFC 7231 datetime format
-        headers['Upload-Expires'] = new Date(
+        responseData.headers['Upload-Expires'] = new Date(
           creation.getTime() + this.store.getExpiration()
         ).toUTCString()
       }
     }
 
-    const writtenRes = this.write(res, 201, {Location: url, ...headers})
+    const writtenRes = this.write(
+      res,
+      responseData.status,
+      Object.assign({Location: url}, responseData.headers),
```

Comment:
I think it should be the other way around so hooks can't override the `Location.
```suggestion
      {...responseData.headers, Location: url},
```

Reply:
As we are aiming at complete control over the response (including the status code), I would let the user override the location header. Possibly, in a customized implementation the resource url is not relevant at all, only a http 200 json response is required.

Reply:
That doesn't make sense because we have `generateUrl` for that, which should be used. It should be locked in here.

Reply:
I see. Initially I thought of being able to unset the location header, but I realized that its still not possible this way. Since location is only valid for 201 and 3xx, it would be good to support this. What do you think?

Reply:
How about conditionally setting it based on the status code?

Reply:
Sounds fine, I pushed the changes